src/paintingTools/brush.js

```diff
@@ -107,13 +107,265 @@ function onImageRendered (e) {
   }
 }
 
+// This method is for fill region of segmentation overlays
+
+function fill (eventData) {
```

Comment:
This function is a bit long and has a lot going on (codeclimate gives it a cognitive complexity of 105). For the sake of maintainability, would it be possible to split this function down into a sequence of appropriately named smaller functions, so that it is easier to follow?

Reply:
I think so! I separated fill function to smaller functions named functionally. 
and these are moved to fill.js.